### ISSUE TITLE
unable run automation scripts on real device

### ISSUE THREAD

**Issue by prathimak** [calabash/calabash-ios#953]
Steps to reproduce issue:
Installed an app on the real device from the xcode and then followed below steps
$ export BUNDLE_ID=xxxx
$ export DEVICE_TARGET=< xxx >
$ export DEVICE_ENDPOINT=http://xxx37265 
$ bundle exec calabash-ios console 
> start_test_server_in_background

App is launched on the device but after 5 mins of waiting i got below error on the console

Unable to start. Make sure you've set APP_BUNDLE_PATH to a build supported by this simulator version
Calabash::Cucumber::Launcher::StartError: Calabash::Cucumber::Launcher::StartError: 

"Timed out waiting for UIAutomation run-loop Error while writing to fifo. RunLoop::Fifo::NoReaderConfiguredError.

env:

xcode version: Version 7.2 (7C68)
calabash version: 'calabash-cucumber', '>= 0.16.3', '< 2.0'

Please let me know if i missing something

**Comment by jmoody** [comment]
Fifo errors on devices can be resolved by reviewing this Hot Topic:

https://github.com/calabash/calabash-ios/wiki/Hot-Topics#7-errnoeintr-interrupted-system-call

Please review this wiki page as well: https://github.com/calabash/calabash-ios/wiki/Testing-on-Physical-Devices

**Comment by prathimak** [comment]
@jmoody thank you for your quick reply, i reviewed hot topic and followed wiki page as well, we are really struck here, is there anything you can suggest me

**Comment by distortia** [comment]
@prathimak, what exactly have you tried from the wiki page?

**Comment by prathimak** [comment]
@distortia from this wiki page https://github.com/calabash/calabash-ios/wiki/Testing-on-Physical-Devices i just followed all the steps using .ipa and .app files when i run the below commands

  export BUNDLE_ID=xxx
$ export DEVICE_TARGET=< UDID >
$ export DEVICE_ENDPOINT=http://<ip>:37265
$ bundle exec cucumber

i received below error
Unable to start. Make sure you've set APP_BUNDLE_PATH to a build supported by this simulator version
 Calabash::Cucumber::Launcher::StartError: 
 
 "Timed out waiting for UIAutomation run-loop Error while writing to fifo. RunLoop::Fifo::NoReaderConfiguredError.

strange is i am using real device not the simulator

if i am wrong can you guide me

**Comment by jmoody** [comment]
You have to through the the FIFO Hot Topic carefully.  When you upgrade Xcode and iOS versions, these FIFO errors are common.

> export DEVICE_ENDPOINT=http://:37265
> curl http://xxxxxx:37265/version

Are you sure you are setting the DEVICE_ENDPOINT correctly?

This means that the app you trying test does not have the server linked properly _or_ there is another app on the device that blocking port 37265 - is there another Calabash app running on the device?

> calabash version: 'calabash-cucumber', '>= 0.16.3', '< 2.0'

Please update to 0.17.0 - this requires a server update as well.

**Comment by prathimak** [comment]
@jmoody thanks for the help, when i tried to run curl http://xxx.xx.xx.xxx:37265/version
curl: (7) Failed to connect to http://xxx.xx.xx.xxx:37265: Network is unreachable, seems like test server is not running

**Comment by distortia** [comment]
Is your target/scheme linked with calabash?

**Comment by prathimak** [comment]
yes my scheme linked with calabash

**Comment by distortia** [comment]
Your computer and phone are both on the same network as well?

**Comment by prathimak** [comment]
yes both are on the same network

**Comment by distortia** [comment]
what version of IOS is the phone?

**Comment by prathimak** [comment]
iPhone 5s , os :8.4.1

**Comment by distortia** [comment]
as @jmoody said, can you try upgrading your calabash-ios to 0.17.0?

**Comment by prathimak** [comment]
ok i can try now

**Comment by TeresaP** [comment]
Are you on a corporate network? Ours blocked network traffic from the ipad unless we either installed MDM files or shared our host machine's wifi and had the iPad connect via that local wifi connection.

**Comment by sapieneptus** [comment]
@prathimak Have you tried this? Run 
```
killall iproxy
iproxy 37265 37265
```
then run the test.

**Comment by jmoody** [comment]
`iproxy` is a process that is _managed_ by run-loop.  It should be automatically killed when the simulator is quit.

* **Simulator: should manage the 'iproxy' process** [#279](https://github.com/calabash/run_loop/pull/279)
* [CoreSimulator.quit_simulator](https://github.com/calabash/run_loop/blob/develop/lib/run_loop/core_simulator.rb#L102)

**Comment by prathimak** [comment]
@TeresaP thanks for your reply i tried at my home network as well similar issue

@sapieneptus  thanks for your reply i tried like that as well

@distortia and @jmoody  i have updated calabash-ios to 0.17.0 and received same error

Unable to start. Make sure you've set APP_BUNDLE_PATH to a build supported by this simulator version
 Calabash::Cucumber::Launcher::StartError: 
 
 "Timed out waiting for UIAutomation run-loop Error while writing to fifo. RunLoop::Fifo::NoReaderConfiguredError.
 
 Logfile: /Users/xxx/.run-loop/results/2015-12-14_11-50-09/run_loop.out
 
 2015-12-14 11:50:10.760 instruments[65625:322503] WebKit Threading Violation - initial use of WebKit from a secondary thread.
 2015-12-14 11:50:11.153 instruments[65625:322532] Attempting to change event horizon while disengage
 Instruments Trace Error : Target failed to run: The operation couldn’t be completed. (FBSOpenApplicationErrorDomain error 4.) : Failed to launch process with bundle identifier 'org.reactjs.native.example.xxxx'
 2015-12-14 11:50:11.153 instruments[65625:322529] Attempting to change event horizon while disengage
 
 
  (Calabash::Cucumber::Launcher::StartError)
 /Users/xxx/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:725:in `new_run_loop'
 /Users/xxx/.rvm/gems/ruby-2.2.1/gems/calabash-cucumber-0.17.0/lib/calabash-cucumber/launcher.rb:580:in `relaunch'
 /Users/xxxx/Documents/Mobile-Automation/components-automation/features/ios_sign_in/support/01_launch.rb:27:in `Before'

**Comment by TeresaP** [comment]
Your BUNDLE_ID doesn't look right. Your bundle identifier is 'org.reactjs.native.example.xxxx'?

**Comment by prathimak** [comment]
@TeresaP  thanks 
I am using app developed using react native

**Comment by TeresaP** [comment]
So you are running their sample app or something? I imagine they would allow you to customize your bundle id for a shipping app so the value looks suspicious to me.

**Comment by distortia** [comment]
Can you try running with app with the calabash-ios console? Maybe it is
something with the cucumber profile or some env/hook issue

**Comment by sapieneptus** [comment]
@prathimak Just as a sanity check, how do you know calabash is linked? 

Basically, in terms of verifying there is a connection:
- If calabash is linked, 
- and `iproxy` is running 37265 37265, 
- and you are using the default calabash server port settings, 
- and the app is running, 
- and your computer is on the same network as the device, 
- and the device is plugged into the computer,
- and you don't have any firewall blocking port 37265, 
- then `curl localhost:37265/version` will return some json.  

Can you verify this?

@jmoody Am I missing any sanity check steps?

**Comment by jmoody** [comment]
> and iproxy is running 37265 37265,

Calabash iOS does not use iproxy.

I updated your comment.

**Comment by jmoody** [comment]
@prathimak 

I don't see that you have gone through all the steps in the Hot Topic about FIFO problems.  You need to do this.

>  $ curl http://xxx.xx.xx.xxx:37265/version

Before you launch, you need to make sure the ipa you are testing _has the server linked_.  If this curl is failing, it means either:

* Your ipa is not linked with Calabash.
* The IP address is not correct.
* The IP address is being blocked.

You might have a look at these tutorials: [How to Add Calabash to Your Project](https://github.com/calabash/calabash-ios/wiki/Tutorial%3A-How-to-add-Calabash-to-Xcode).

This might also be helpful:  https://github.com/calabash/calabash-ios/wiki/Improving-Network-Stability

As a sanity check, I have a colleague who is building a ReactJS iOS app and he is using Calabash.

**Comment by prathimak** [comment]
@jmoody thanks for your reply now when i do the curl it returns json, i am not using ipa i am using .app file

**Comment by prathimak** [comment]
i could see in xcode out 2015-12-15 14:03:08.474 DEBUG CalabashServer:242 | App Base SDK: iphoneos9.2
2015-12-15 14:03:08.475 DEBUG CalabashServer:274 | IPHONE_SIMULATOR_ROOT: (null)
2015-12-15 14:03:08.483 DEBUG CalabashServer:260 | Calabash iOS server is listening on: 192.168.2.10 port 37265

**Comment by jerseysu** [comment]
@prathimak Follow you steps, I have the similar problem like you the APP launched but takes time to wait (No response).

After set the environment parameter DEVICE and DEVICE_TARGET as same UniqueDeviceID then solved this problem. Perhaps this solution can help to solved the problem.

```
:device => "udid",
:device_target => "udid",
```

**Comment by jmoody** [comment]
@jerseysu `:device` is not respected.

The other reason FIFO errors can happen is if your application has been signed by a Distribution profile.  Apps must be signed by a Developer cert.

**Comment by jmoody** [comment]
I am closing because there has been no response.

My best guess is that:

1. You signed your application with a Distribution certificate.
2. You did not link Calabash with your application.

**Comment by jmoody** [comment]
I have released run_loop 2.0.10.pre1 which might resolve the FIFO problems.  Thanks @TeresaP
